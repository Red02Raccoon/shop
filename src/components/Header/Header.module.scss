@import 'styles/_variables.scss';

.header {
    @apply flex justify-between items-center py-4 px-8 border-slate-100 border-b-2 bg-white fixed w-full top-0 left-0 z-10;
}

.logo {
    background: $primary-color;
    color: #fff;
    padding: 10px;
}
