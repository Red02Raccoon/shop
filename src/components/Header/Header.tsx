import { useNavigate } from 'react-router-dom';
import { NavBar } from '../NavBar';

import styles from './Header.module.scss';

const Header = () => {
    const navigate = useNavigate();

    const handleLogoClick = () => {
        navigate('/');
    };

    return (
        <div className={styles.header}>
            <div className="logo" data-testid="logo" onClick={handleLogoClick}>
                Header
            </div>

            <NavBar />
        </div>
    );
};

export default Header;
