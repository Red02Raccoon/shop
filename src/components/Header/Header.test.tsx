import { ReactNode } from 'react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';

import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import Header from './Header';

function renderWithRouter(component: ReactNode) {
    const history = createMemoryHistory();

    return {
        ...render(
            <Router location={history.location} navigator={history}>
                {component}
            </Router>
        ),

        history,
    };
}

describe('<Header/>', () => {
    it('should display correct logo text', () => {
        renderWithRouter(<Header />);

        expect(screen.getByTestId('logo').textContent).toEqual('Header');
    });

    it('should navigate to root page after pressing logo', async () => {
        const { history } = renderWithRouter(<Header />);

        await userEvent.click(screen.getByTestId('logo'));

        expect(history.location.pathname).toBe('/');
    });
});
