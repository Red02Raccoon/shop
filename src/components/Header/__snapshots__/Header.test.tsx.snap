// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> Logo should match the snapshot: additional snapshot description 1`] = `
<div
  class="logo"
  data-testid="logo"
>
  Logo
</div>
`;
